clients/src/test/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumerTest.java

```diff
@@ -169,6 +184,35 @@ private AsyncKafkaConsumer<String, String> newConsumer(ConsumerConfig config) {
         );
     }
 
+    private AsyncKafkaConsumer<String, String> newConsumer(
+        FetchBuffer fetchBuffer,
+        ConsumerInterceptors<String, String> interceptors,
+        ConsumerRebalanceListenerInvoker rebalanceListenerInvoker,
+        SubscriptionState subscriptions,
+        List<ConsumerPartitionAssignor> assignors,
+        String groupId,
+        String clientId) {
+        return new AsyncKafkaConsumer<>(
```

Comment:
@philipnee I removed this on purpose from the test file to use the normal constructor and not just test a mock construction of the class. I'm sure there could have been a better way than to reintroduce the constructor that I remove. However, since we are otherwise not converging with this PR, I am going to merge this. Please consider following up with a PR that removes this constructor again